gcp_variant_transforms/beam_io/vcfio.py

```diff
@@ -453,180 +500,209 @@ class _VcfSource(_TextSource):
-            'Invalid record in VCF file %s. Error: %s' % (file_name, str(e)))
-
-  def _convert_to_variant_record(self, record, infos, formats):
-    """Converts the PyVCF record to a :class:`Variant` object.
-
-    Args:
-      record (:class:`~vcf.model._Record`): An object containing info about a
-        variant.
-      infos (dict): The PyVCF dict storing INFO extracted from the VCF header.
-        The key is the info key and the value is :class:`~vcf.parser._Info`.
-      fromats (dict): The PyVCF dict storing FORMAT extracted from the VCF
-        header. The key is the FORMAT key and the value is
-        :class:`~vcf.parser._Format`.
-    Returns:
-      A :class:`Variant` object from the given record.
-    """
-    variant = Variant()
-    variant.reference_name = record.CHROM
-    variant.start = record.start
-    variant.end = record.end
-    variant.reference_bases = (
-        record.REF if record.REF != MISSING_FIELD_VALUE else None)
-    # ALT fields are classes in PyVCF (e.g. Substitution), so need convert them
-    # to their string representations.
-    variant.alternate_bases.extend(
-        [str(r) for r in record.ALT if r] if record.ALT else [])
-    variant.names.extend(record.ID.split(';') if record.ID else [])
-    variant.quality = record.QUAL
-    # PyVCF uses None for '.' and an empty list for 'PASS'.
-    if record.FILTER is not None:
-      variant.filters.extend(record.FILTER if record.FILTER else [PASS_FILTER])
-    for k, v in record.INFO.iteritems():
-      # Special case: END info value specifies end of the record, so adjust
-      # variant.end and do not include it as part of variant.info.
-      if k == END_INFO_KEY:
-        variant.end = v
-        continue
-      field_count = None
-      if k in infos:
-        field_count = self._get_field_count_as_string(infos[k].num)
-      variant.info[k] = VariantInfo(data=v, field_count=field_count)
-
-    for sample in record.samples:
-      call = VariantCall()
-      call.name = sample.sample
-      for allele in sample.gt_alleles or [MISSING_GENOTYPE_VALUE]:
-        if allele is None:
-          allele = MISSING_GENOTYPE_VALUE
-        call.genotype.append(int(allele))
-      phaseset_from_format = (getattr(sample.data, PHASESET_FORMAT_KEY)
-                              if PHASESET_FORMAT_KEY in sample.data._fields
-                              else None)
-      # Note: Call is considered phased if it contains the 'PS' key regardless
-      # of whether it uses '|'.
-      if phaseset_from_format or sample.phased:
-        call.phaseset = (str(phaseset_from_format) if phaseset_from_format
-                         else DEFAULT_PHASESET_VALUE)
-      for field in sample.data._fields:
-        # Genotype and phaseset (if present) are already included.
-        if field in (GENOTYPE_FORMAT_KEY, PHASESET_FORMAT_KEY):
+        raise ValueError('Invalid record in VCF file. Error: %s' % str(e))
+
+    def _convert_to_variant_record(self, record, infos, formats):
```

Comment:
To confirm: no changes were actually made to this logic? Not sure why git can't figure out that there are actually no diffs :)

Reply:
No changes were made to the logic, but the function itself was moved from _VcfSource to _VcfRecordIterator.